tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,346 @@
+                }
+
+            }
+            operations.addAndGet(iteration % performanceUpdateFrequency);
+        }
+    }
+
+    public Employee createEmployee(int id) {
+        switch (serializationStrategy) {
+            case DATA_SERIALIZABLE:
+                return new DataSerializableEmployee(id);
+            case JAVA_SERIALIZABLE:
+                return new SerializableEmployee(id);
+            case IDENTIFIED_DATA_SERIALIZABLE:
+                return new IdentifiedDataSerializableEmployee(id);
+            case PORTABLE:
+                return new PortableEmployee(id);
+        }
+        return null;
+    }
+
+    //Custom Predicates
+    private static class NamePredicate implements Predicate<String, Employee> {
+        private final String name;
+
+        private NamePredicate(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean apply(Map.Entry<String, Employee> entry) {
+            Employee value = entry.getValue();
+            return value.getName().equals(name);
+        }
+    }
+
+    public interface Employee {
+
+        public String getName();
+    }
+
+    public static class DataSerializableEmployee implements DataSerializable,Employee {
+
+        protected int id;
+        protected String name;
+
+        public String getName() {
+            return name;
+        }
+
+        public DataSerializableEmployee() {
+        }
+
+        public DataSerializableEmployee(int id) {
+            this.id = id;
+            name = names[random.nextInt(names.length)];
+        }
+
+        @Override
+        public void writeData(ObjectDataOutput objectDataOutput) throws IOException {
+            objectDataOutput.writeInt(id);
+            objectDataOutput.writeUTF(name);
+        }
+
+        @Override
+        public void readData(ObjectDataInput objectDataInput) throws IOException {
+            id = objectDataInput.readInt();
+            name = objectDataInput.readUTF();
+        }
+
+        @Override
+        public String toString() {
+            return "Employee{" +
+                    "id=" + id +
+                    ", name='" + name + '\'' +
+                    '}';
+        }
+    }
+
+    public static class EmployeeDataSerializableFactory implements DataSerializableFactory{
+
+        public static final int FACTORY_ID = 3000;
+
+        @Override
+        public IdentifiedDataSerializable create(int i) {
+            return new IdentifiedDataSerializableEmployee();
+        }
+    }
+
+
+    public static class IdentifiedDataSerializableEmployee implements IdentifiedDataSerializable,Employee {
+        protected int id;
+        protected String name;
+
+        public String getName() {
+            return name;
+        }
+
+        public IdentifiedDataSerializableEmployee() {
+        }
+
+        public IdentifiedDataSerializableEmployee(int id) {
+            this.id = id;
+            name = names[random.nextInt(names.length)];
```

Comment:
Can you move the selection of the random name into the warmup method? So the constructor should look like:
public IdentifiedDataSerializableEmployee(int id,String name){...}